app/controllers/api/service_dialogs_controller.rb

```diff
@@ -1,6 +1,13 @@
 module Api
   class ServiceDialogsController < BaseController
-    before_action :set_additional_attributes, :only => [:show]
+    def show
+      if params[:c_id]
+        resource = Dialog.find(params[:c_id])
+        render :json => single_resource(resource, params).target!
+      else
+        render :json => dialog_collection(params).target!
```

Comment:
are the elements returned with id from serializer primary elements in the API ? (i.e. collections like /api/dialogs/), if so would be nice if the renderer could add the API hrefs for those.